packages/curriculum-compiler-json/compilers/helpers.js

```diff
@@ -0,0 +1,36 @@
+const { getCompiler } = require('@enkidevs/curriculum-compiler-string')
+const { contentTypes } = require('@enkidevs/curriculum-helpers')
+
+function getInsightCompiler () {
+  return getCompiler(contentTypes.INSIGHT)
+}
+function compileNodeToInsightMarkdown (node) {
+  if (!node || !node.children) {
+    throw new Error('Cannot compile invalid node')
+  }
+  return getInsightCompiler().compileSync({
+    type: 'root',
+    children: node.children
+  })
+}
+
+function getAnswersFromNode (node) {
+  const answersASTList = node.children.find(
+    child => child.type === 'list' && child.answers
+  )
+  if (!answersASTList || !answersASTList.children) {
```

Comment:
A slightly safer check for `children` everywhere could be to use `!isArray` instead of just checking for falsyness

Reply:
thing is `.isArray(children)` is `true` for empty arrays while checking the `children` directly will return `false` for empty arrays

Reply:
actually stumbled accross this issue on the content consumer:

there was a check similar to this:
```
!Array.isArray(otherFiles) ? false : otherFiles.filter().etc()
```

This would hit `otherFiles.filter...` either when `otherFiles` was `[]` or when it was populated

Reply:
I think `Array.isArray` is useful when you can get other data types rather than `array` on that specific variable. But given all this is following the unist AST paradigm, think `children` will be an array everytime.

But it's true another saftey check won't hurt